src/main/scala/org/apache/spark/sql/rikai/Image.scala

```diff
@@ -28,21 +28,33 @@ private[spark] class ImageType extends UserDefinedType[Image] {
   override def sqlType: DataType =
     StructType(
       Seq(
-        StructField("uri", StringType, false)
+        StructField("data", BinaryType, true),
+        StructField("uri", StringType, true)
       )
     )
 
   override def pyUDT: String = "rikai.spark.types.vision.ImageType"
 
   override def serialize(obj: Image): Any = {
-    val row = new GenericInternalRow(1);
-    row.update(0, UTF8String.fromString(obj.uri))
+    val row = new GenericInternalRow(2);
+    obj.data.map(x => row.update(0, x))
+    obj.uri.map(x => row.update(1, UTF8String.fromString(x)))
     row
   }
 
   override def deserialize(datum: Any): Image =
     datum match {
-      case row: InternalRow => new Image(row.getString(0).toString())
+      case row: InternalRow =>
+        new Image(
+          row.isNullAt(0) match {
```

Comment:
does `Option.unless(row.isNullAt(0))(row.getBinary(0))` and `Option.when(row.isNullAt(1))(row.getString(1).toString)` work?

Reply:
lemme try some scala magic

Reply:
Hmm, in. scala 2.12 `Option` does not have this method